### PR TITLE
Pyscf: 1.7.3 -> 1.7.6

### PR DESCRIPTION
**Pull request by sheepforce**
A few more experiments on PySCF. Besides an update, I also added a batsTest case and XCFun support (in the hope that it would change something). This still does not work for me. I have added an alternative build style using cmake instead of setup.py tools. When the tests are disabled in the setup.py version they give identical behaviour. An unfortunate observation; not a single example from the PySCF examples runs correctly (SCF never converges or matrices become singular). The added example is such a case. Interestingly the SCF basically converges when looking at the numerical values. BLAS has a minor effect on the number of cycles but down to numerical accuracy the results are the same.

Nevertheless; it still does not work. @markuskowa Can you build PySCF or do you get the same AM1 error as mentioned in #29 now?

**PySCF does not work in this branch and in master for me. Maybe do not merge yet**

### CODE REVIEW

**Comment by sheepforce** [comment]
Phew, I'm running out of ideas. The SCF even converges in the test case?
I just also tried PySCF again on OpenSuse multi-user Nix and still get a different error than on NixOS (NixOS still fails with the AM1 assertion errors from #29 ). On OpenSuse it is
```
running test
WARNING: Testing via this command is deprecated and will be removed in a future version. Users looking for a generic test entry point independent of test runner are encouraged to use tox.
running egg_info
writing pyscf.egg-info/PKG-INFO
writing dependency_links to pyscf.egg-info/dependency_links.txt
writing requirements to pyscf.egg-info/requires.txt
writing top-level names to pyscf.egg-info/top_level.txt
reading manifest file 'pyscf.egg-info/SOURCES.txt'
reading manifest template 'MANIFEST.in'
warning: no files found matching 'AUTHORS'
no previously-included directories found matching 'pyscf/lib/build'
warning: no files found matching '*.so*' under directory 'pyscf/lib/deps'
warning: no files found matching '*.dylib*' under directory 'pyscf/lib/deps'
warning: no files found matching 'pyscf/lib/*.so*'
warning: no files found matching 'pyscf/lib/*.dylib*'
warning: no previously-included files matching '*.h' found under directory 'pyscf/lib'
warning: no previously-included files matching '*.cl' found under directory 'pyscf/lib'
writing manifest file 'pyscf.egg-info/SOURCES.txt'
running build_ext
copying build/lib.linux-x86_64-3.8/pyscf/lib/libnp_helper.so -> pyscf/lib
copying build/lib.linux-x86_64-3.8/pyscf/lib/libcgto.so -> pyscf/lib
copying build/lib.linux-x86_64-3.8/pyscf/lib/libcvhf.so -> pyscf/lib
copying build/lib.linux-x86_64-3.8/pyscf/lib/libao2mo.so -> pyscf/lib
copying build/lib.linux-x86_64-3.8/pyscf/lib/libcc.so -> pyscf/lib
copying build/lib.linux-x86_64-3.8/pyscf/lib/libfci.so -> pyscf/lib
copying build/lib.linux-x86_64-3.8/pyscf/lib/libmcscf.so -> pyscf/lib
copying build/lib.linux-x86_64-3.8/pyscf/lib/libri.so -> pyscf/lib
copying build/lib.linux-x86_64-3.8/pyscf/lib/libhci.so -> pyscf/lib
copying build/lib.linux-x86_64-3.8/pyscf/lib/libpbc.so -> pyscf/lib
copying build/lib.linux-x86_64-3.8/pyscf/lib/libmbd.so -> pyscf/lib
copying build/lib.linux-x86_64-3.8/pyscf/lib/libdft.so -> pyscf/lib
copying build/lib.linux-x86_64-3.8/pyscf/lib/libxc_itrf.so -> pyscf/lib
/build/source/pyscf/lib/misc.py:46: H5pyDeprecationWarning: Using default_file_mode other than 'r' is deprecated. Pass the mode to h5py.File() instead.
  h5py.get_config().default_file_mode = 'a'
error: no file with expected extension
builder for '/nix/store/nap83x0pwspk7pdif7x14wrahaiwgq60-python3.8-pyscf-1.7.6.drv' failed with exit code 1
error: build of '/nix/store/nap83x0pwspk7pdif7x14wrahaiwgq60-python3.8-pyscf-1.7.6.drv' failed
```

How can this be so different? :neutral_face:

**Comment by markuskowa** [comment]
> The SCF even converges in the test case?

I guess not. This is actually something that the test case should catch.

```
SCF not converged.
SCF energy = 2259.33331198504
CASSCF energy = 1022.09546638825
CASCI E = 1022.09546638825  E(CI) = 217.291978257606  S^2 = 0.0000000
CASCI state-averaged energy = 1022.09546638825
CASCI energy for each state
  State 0 weight 0.5  E = 1019.75159904634 S^2 = 0.0000000
  State 1 weight 0.5  E = 1024.43933373017 S^2 = 0.0000000

WARN: Mulitple states found in CASCI solver. First state is used to compute the natural orbitals in active space.

CASCI state 0  E = 1069.0797005669  E(CI) = 264.276212436254  S^2 = 0.0000000
CASCI state 1  E = 1069.36164442223  E(CI) = 264.558156291581  S^2 = 0.0000000
CASCI state 2  E = 1080.04664091927  E(CI) = 275.243152788616  S^2 = 2.0000000
CASCI state 3  E = 1080.17636561577  E(CI) = 275.372877485117  S^2 = 0.0000000
Sr    (-1)',   E = -189.09606584262485
Si    (+1)',   E = -273.20070405251954
Sijrs (0)  ,   E = 1041.02681257069798
Sijr  (+1) ,   E = -1358.48655661754856
Srsi  (-1) ,   E = 46.21754417708298
Srs   (-2) ,   E = 0.05401918703554
Sij   (+2) ,   E = -218.41802109232862
Sir   (0)' ,   E = -2456.51889103192480
Nevpt2 Energy = -3408.421862702130056
Total energy of first excited state -2339.0602182799003
```

**Comment by sheepforce** [comment]
> This is actually something that the test case should catch.

I will take care of it as soon as it builds correctly at all :slightly_smiling_face:

File: pkgs/lib/libcint/default.nix [hunk 1]
```diff
@@ -1,8 +1,8 @@
 { lib, stdenv, fetchFromGitHub, cmake
-, openblas, python
+, blas, python
 } :
 let
-  version = "3.0.15";
```
Comment (markuskowa) [inline pkgs/lib/libcint/default.nix:5]: This libcint3, which is(?)/was the  legacy version for pyscf. There is `libcint-4.4.0` is in upstream nixpkgs. I think we can completely remove libcint3 here?

File: pkgs/lib/xcfun/default.nix [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    repo = pname;
+    rev = "v${version}";
+    sha256= "1bj70cnhbh6ziy02x988wwl7cbwaq17ld7qwhswqkgnnx8rpgxid";
+  };
+
+  meta = with lib; {
+    description = "A library of exchange-correlation functionals with arbitrary-order derivatives.";
+    homepage = "https://github.com/dftlibs/xcfun";
+    license = licenses.mpl20;
+    platforms = platforms.unix;
```
Comment (markuskowa) [inline pkgs/lib/xcfun/default.nix:43]:
```suggestion
    platforms = platforms.linux;
```

File: pkgs/apps/pyscf/cmake.nix [hunk 3]
```diff
@@ -0,0 +1,74 @@
+{ stdenv, lib, toPythonModule, cmake, fetchFromGitHub, writeTextFile, libcint, libxc, xcfun, blas,
```
Comment (markuskowa) [inline pkgs/apps/pyscf/cmake.nix:1]: Is that whole file still needed?

**Review by markuskowa — CHANGES_REQUESTED** [review]

**Review by markuskowa — APPROVED** [review]

**Pull request merged**